tests/test.js

```diff
@@ -438,4 +438,108 @@ test('NbsDecoder.getPackets() returns the last packet of each type when given a
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0} , 1);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, testArray , 1);
+
+  assert.ok(
+   tsToBigInt(singleTypeTimestamp) > tsToBigInt(start),
+   'The new return Nbs timestamp is greater than the input timestamp'
+ );
+
+ assert.ok(
+  tsToBigInt(multiTypeTimestamp) > tsToBigInt(start),
+  'The new return Nbs timestamp is greater than the input timestamp'
+);
+ })
+// });
+
+/* Check stepping forward by five returns the expected timestamp.  */
+test(`NbsDecoder.nextTimestamp() returns exact value expected`, () => {
+  const [start, end] = decoder.getTimestampRange( { type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(start, { type: pingType, subtype: 0 }, 5);
+  const multiTypeTimestamp = decoder.nextTimestamp(start, testArray , 5);
+
+  assert.equal(tsToBigInt(singleTypeTimestamp),
+  tsToBigInt({seconds: 1015, nanos: 0}))
+
+  assert.equal(tsToBigInt(multiTypeTimestamp),
+  tsToBigInt({seconds: 1012, nanos: 0}))
+});
+
+/* Check stepping backward returns a timestamp which is less than input. */
+test('NbsDecoder.nextTimestamp() returns a timestamp less than input', () => {
+  const [start, end] = decoder.getTimestampRange({ type: pingType, subtype: 0 });
+  const singleTypeTimestamp = decoder.nextTimestamp(end, { type: pingType, subtype: 0 }, -1)
+
+   const multiTypeTimestamp = decoder.nextTimestamp(end, testArray, -1)
+
+  assert.ok(
+   tsToBigInt(singleTypeTimestamp) < tsToBigInt(end),
+   'The return Nbs timestamp is less than the input timestamp for a single type'
+ );
+
+ assert.equal(
+  tsToBigInt(singleTypeTimestamp), tsToBigInt({seconds: 1894, nanos: 0}));
+  'The return Nbs timestamp exactly matches the expected timestamp'
+
+
+  assert.ok(
+    tsToBigInt(multiTypeTimestamp) < tsToBigInt(end),
+    'The return Nbs timestamp is less than the input timestamp for multiple types'
+  );
+});
+
+/* Error checking. */
+test('NbsDecoder.nextTimestamp() Throws error on missing or incorrect inputs.', () => {
+  assert.throws(
+    () => {
+      decoder.nextTimestamp();
+    },
+    /invalid type for argument `timestamp`: expected positive number or BigInt/,
+    'NbsDecoder.nextTimestamp() throws for missing `timestamp` argument'
+  );
+
+  assert.throws(
+    () => {
+      decoder.nextTimestamp("(1500 * 1e9)");
+    },
+    /invalid type for argument `timestamp`: expected positive number or BigInt/,
+    'NbsDecoder.nextTimestamp() throws for invalid `timestamp` argument'
+  );
+
+  assert.throws(
+    () => {
+      decoder.nextTimestamp(0, { type: pingType, subtype: "0" }, 1);
+    },
+    /invalid type for argument `typeSubtype`: invalid `.subtype`: expected number/,
+    'NbsDecoder.nextTimestamp() throws for object with invalid `subtype`'
+  );
+
+  assert.throws(
+    () => {
+      decoder.nextTimestamp((1500 * 1e9), "a");
+    },
+    /invalid step input. A number was expected./,
```

Comment:
To match the phrasing of the other error message:

```suggestion
    /invalid type for argument `step`: expected number/,
```

Reply:
Refactored error message.